src/hardware/sblaster.cpp

```diff
@@ -2122,69 +2122,64 @@ static void SBLASTER_CallBack(uint32_t len)
 	}
 }
 
-SB_TYPES find_sbtype()
+static std::optional<SB_TYPES> parse_sbtype_pref(const std::string& pref)
 {
-	const auto sect = static_cast<Section_prop *>(control->GetSection("sblaster"));
-	assert(sect);
+	if (pref == "gb") {
+		return SBT_GB;
 
-	const std::string pref = sect->Get_string("sbtype");
+	} else if (pref == "sb1") {
+		return SBT_1;
 
-	// Default
-	auto sbtype = SB_TYPES::SBT_NONE;
+	} else if (pref == "sb2") {
+		return SBT_2;
 
-	// Newest to oldest
-	if (pref == "sb16") {
-		sbtype = SBT_16;
-	} else if (pref == "sbpro2") {
-		sbtype = SBT_PRO2;
 	} else if (pref == "sbpro1") {
-		sbtype = SBT_PRO1;
-	} else if (pref == "sb2") {
-		sbtype = SBT_2;
-	} else if (pref == "sb1") {
-		sbtype = SBT_1;
-	} else if (pref == "gb") {
-		sbtype = SBT_GB;
+		return SBT_PRO1;
+
+	} else if (pref == "sbpro2") {
+		return SBT_PRO2;
+
+	} else if (pref == "sb16") {
+		return SBT_16;
 	}
```

Comment:
Should we add a call to parse_bool_setting to also support "none" here? Having no Sound/Game Blaster should still be an option.

```
	} else if (const auto maybe_bool = parse_bool_setting(pref); maybe_bool) {
		if (!*maybe_bool) {
			return SBT_NONE;
		}
	}
```

Reply:
Oh, thanks for catching this. This would have introduced the same regression for `sbtype` that I'm fixing for `oplmode` 🤦🏻 Fixing 😅